### ISSUE TITLE
Add support to pass additional argument to enable private channel communication 

### ISSUE THREAD

**Issue by shivamsupr** [ryanwinchester/slack_elixir#5]
Currently, `channel_server.ex` fetches only public channels using the below code
```elixir
defp fetch_channels(token) do
    "users.conversations"
    |> Slack.API.stream(token, "channels")
    |> Enum.map(& &1["id"])
end
```

There should be an option to pass the `types` argument to the `fetch_channels` func so that the bot can access even private channels.
```elixir
# channel_types: "public_channel, private_channel"

defp fetch_channels(token, channel_types) do
    "users.conversations"
    |> Slack.API.stream(token, "channels", types: channel_types)
    |> Enum.map(& &1["id"])
end
```

Also, since everything is config-driven, this should be allowed at the config level. Something like this but via `runtime.exs`
```elixir
config = [
      app_token: "xapp-your-app-token",
      bot_token: "xoxb-your-bot-token",
      bot: MyApp.Slackbot,
      channel_types: "public_channel, private_channel"
    ]
```

**Comment by ryanwinchester** [comment]
Fixed in 02ecbe1585be6e59db364a25035184b5dda337db

Released `v1.1.0`

By default it joins all conversations it is in, that it has access to.

You can customize it with the config:

```elixir
config :my_app, MyApp.SlackBot,
  channels: [
    types: ["public_channel", "im"]
  ],
  # ...etc
```

**Comment by shivamsupr** [comment]
This is awesome, thanks @ryanwinchester
